src/json2netns/route.py

```diff
@@ -0,0 +1,103 @@
+from ipaddress import ip_address, ip_network
+from typing import Dict, List
+import logging
+from subprocess import check_output
+
+from json2netns.consts import DEFAULT_IP
+
+
+LOG = logging.getLogger(__name__)
+
+
+class Route:
+    IP = DEFAULT_IP
+
+    def __init__(self, netns_name: str, name: str, route: Dict) -> None:
+        self.name = name
+        self.netns_name = netns_name
+        self.route = route
+
+    def __validate_routes(self, route_test: Dict) -> int:
+        """Check to make sure all elements exist that are needed for a valid route, then validate those elements"""
+        if (route_test["dest_prefix"]) and (
+            route_test["next_hop_ip"] or route_test["exit_if_name"]
+        ):
+            # Check for bad destination prefix, if so skip route installation
+            try:
+                ip_network(route_test["dest_prefix"])
+            except ValueError:
+                LOG.error(
+                    f"{route_test['dest_prefix']} is not a valid network address, please check config"
+                )  # convert to error log
+                return 1
+            # Check if a next hop IP if specified, if so check for bad next hop address and skip route installation
+        if route_test["next_hop_ip"]:
+            try:
+                ip_address(route_test["next_hop_ip"])
+            except ValueError:
+                LOG.error(
+                    f"{route_test['next_hop_ip']} is not a valid ip address, please check config"
+                )  # convert to error log
+                return 1
+        return 0
+
+    def __route_exists(self, route_prefix: str) -> bool:
+        """Checks if route exists already (maintain idempotency)"""
+        route_cmd = [
+            self.IP,
+            "netns",
+            "exec",
+            self.netns_name,
+            self.IP,
+            "route",
+            "show",
+            route_prefix,
+        ]
+        cp = (check_output(route_cmd)).decode("utf-8")
+        if cp == "":
+            return 0
```

Comment:
Can we use return code rather than output please.
Reason: Much more likely one day this might include output or change behavior here but return code should be consistent.

Reply:
Yeah the problem is the return code from that command is 0 even if the route isn't in the table.  It should be possible to grep the table and get a return code that way

Reply:
Using pipe into another run was cumbersome without enabling unsafe shell parameter. Used built in str find method on the output which gives return values.